src/core/plugin/luapi_application.h

```diff
@@ -1402,6 +1403,43 @@ static int applib_changeCurrentPageBackground(lua_State* L) {
     return 0;
 }
 
+/**
+ * Query all colors of the current palette.
+ *
+ * Example1: palette = app.getColorPalette()
+ * possible return value:
+ * {
+ *     [1] = {color = 0xffffff, name = "white" },
+ *     [2] = {color = 0x000000, name = "black" },
+ *     [3] = {color = 0xdc8a78, name = "Rosewater" },
+ * }
+ *
+ */
+static int applib_getColorPalette(lua_State* L) {
+    // discard any extra arguments passed in
+    lua_settop(L, 0);
+
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Settings* settings = plugin->getControl()->getSettings();
+    const Palette& palette = settings->getColorPalette();
+
+
+    lua_newtable(L);  // return table
+    for (size_t i{0}; i < palette.size(); i++) {
+        lua_pushinteger(L, static_cast<lua_Integer>(i) + 1);  // index in return table
```

Comment:
For consistency: 
```suggestion
    for (size_t i = 1; i <= palette.size(); i++) {
        lua_pushinteger(L, as_signed(i));  // index in return table
```

Reply:
I think `i < palette.size()` was correct. At `i = palette.size()` the wrap around will already happen.

Reply:
I missed that you would have to use `palette.getColorAt(i-1)` then.